src/wallet/rpcwallet.cpp

```diff
@@ -3546,22 +3545,23 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
     }
 
     int start_height = 0;
-    uint256 start_block, stop_block;
+    Optional<int> stop_height;
+    uint256 start_block;
     {
         auto locked_chain = pwallet->chain().lock();
-        Optional<int> tip_height = locked_chain->getHeight();
+        LOCK(pwallet->cs_wallet);
```

Comment:
Do you need to hold the wallet lock for this entire block? Does it make sense to call:

```
WITH_LOCK(pwallet->cs_wallet, pwallet->GetLastBlockHeight());
```